RSDK-9091,RSDK-9088 Emit logs as they come from setup phase command

maximpertsov opened the pull request:
### Changes

Emit logs of stdio/stderr messages from setup phase command as they come instead of only at the end. ~Combined output is re-logged after the command completes.~ EDIT: [not anymore](https://github.com/viamrobotics/rdk/pull/4475#discussion_r1809701738)

Also fix bug where script logs were not being posted to app.viam

### Demo

Example run with a setup phase that contains a mongo pull command:

```
2024-10-23T17:17:15.667Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1098      executing first run script        {"module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:16.165Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"6: Pulling from library/mongo","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:16.276Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"6414378b6477: Pulling fs layer","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:16.276Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"d759adb89a6b: Pulling fs layer","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:16.276Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"ba36503b8fbe: Pulling fs layer","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:16.276Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"ce3aea978520: Pulling fs layer","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:16.276Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"4afd0f756edf: Pulling fs layer","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:16.276Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"3f39e6c4bb86: Pulling fs layer","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:16.276Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"575208bc5436: Pulling fs layer","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:16.276Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"1588a03ab798: Pulling fs layer","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:16.276Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"ce3aea978520: Waiting","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:16.276Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"1588a03ab798: Waiting","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:16.276Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"575208bc5436: Waiting","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:16.276Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"4afd0f756edf: Waiting","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:16.276Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"3f39e6c4bb86: Waiting","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:16.467Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"d759adb89a6b: Verifying Checksum","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:16.467Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"d759adb89a6b: Download complete","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:17.012Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"ce3aea978520: Verifying Checksum","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:17.012Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"ce3aea978520: Download complete","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:17.179Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"ba36503b8fbe: Download complete","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:17.407Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"4afd0f756edf: Verifying Checksum","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:17.407Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"4afd0f756edf: Download complete","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:17.536Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"3f39e6c4bb86: Verifying Checksum","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:17.536Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"3f39e6c4bb86: Download complete","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:17.960Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"1588a03ab798: Verifying Checksum","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:17.960Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"1588a03ab798: Download complete","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:25.195Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"6414378b6477: Verifying Checksum","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:25.195Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"6414378b6477: Download complete","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:25.956Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"6414378b6477: Pull complete","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:26.040Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"d759adb89a6b: Pull complete","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:26.158Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"ba36503b8fbe: Pull complete","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:26.205Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"ce3aea978520: Pull complete","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:26.228Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"4afd0f756edf: Pull complete","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:26.254Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"3f39e6c4bb86: Pull complete","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:44.897Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"575208bc5436: Verifying Checksum","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:44.897Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"575208bc5436: Download complete","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:48.885Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"575208bc5436: Pull complete","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:48.909Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"1588a03ab798: Pull complete","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:48.920Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"Digest: sha256:8192e949e2d5d02be195f96afef1cf30adc87a4d5e631895497e11f4acc7a429","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:48.924Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"Status: Downloaded newer image for mongo:6","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:48.926Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"docker.io/library/mongo:6","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:48.930Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"-------------------------------------","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:48.930Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"Congratulations!","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:48.930Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:48.930Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"The setup script ran successfully!","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:48.930Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:48.930Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"This message is obnoxiously large for","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:48.930Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"testing purposes.","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:48.930Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:48.930Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"Sincerely,","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:48.930Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"First Run Script","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:48.930Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1137      got stdio       {"output":"-------------------------------------","module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
2024-10-23T17:17:48.930Z        INFO    rdk.modmanager.first_run        modmanager/manager.go:1166      first run script succeeded      {"module":"5784b2a8-d84f-473b-9be5-9a2abe7e8752_my_setup_phase","path":"/home/mp/.viam/packages/data/module/5784b2a8-d84f-473b-9be5-9a2abe7e8752-my_setup_phase-0_0_13-linux-amd64/first_run.sh"}
```

dgottlieb commented on module/modmanager/manager.go:
```diff
@@ -1089,11 +1090,12 @@ func (mgr *Manager) FirstRun(ctx context.Context, conf config.Module) error {
 	}
 
-	logger.Info("executing first run script")
+	logger = logger.With("path", firstRunPath)
```
I love including an identifier like this so someone can grep for all of the logs pertaining to a single run.

dgottlieb commented on module/modmanager/manager.go:
```diff
@@ -1119,15 +1121,55 @@ func (mgr *Manager) FirstRun(ctx context.Context, conf config.Module) error {
+		logger.Errorw("first run failed", "error", err, "combined output", combined)
+		return err
+	}
+	logger.Infow("first run succeeded", "combined output", combined)
```
Because you attached a unique identifier to each process run, I don't think this is necessary. Incidentally, this log can also be problematic. This would get put into the database as a single document and be subject to the 16MB limit.

maximpertsov replied:
good point, removing

cheukt commented on module/modmanager/manager.go:
```diff
@@ -1119,15 +1121,55 @@ func (mgr *Manager) FirstRun(ctx context.Context, conf config.Module) error {
+	go func() {
+		for scanErr.Scan() {
+			out := scanErr.Text()
+			logger.Warnw("first run stderr", "output", out)
```
why warn and not just err?

dgottlieb replied:
Warn felt appropriate to me. All kind of "not errors" go to stderr. And so long as the script returns an exit code of 0, we're going to mark it as a success. It might be surprising if there was a bunch of "ERROR" logs and that end up as a successful run.

maximpertsov replied:
yep, @dgottlieb's comment pretty much summarizes my thoughts on this. I can add a code comment with my rationale if that helps

dgottlieb commented on module/modmanager/manager.go:
```diff
@@ -1119,15 +1121,55 @@ func (mgr *Manager) FirstRun(ctx context.Context, conf config.Module) error {
 
-	resultLogger.Infow("command succeeded")
+	scanOut := bufio.NewScanner(stdOut)
+	go func() {
```
I know there are multiple stakeholders here, but I'm perfectly happy avoiding managed process.

I'm not even sure we can* use managed process here without changes. I don't think it exposes the equivalent of `cmd.Wait`. Which we need to know if the process exited with a 0 or non-zero exit code.

dgottlieb replied:
managed process API: https://github.com/viamrobotics/goutils/blob/56b97fca11d21441a50b51bc2b75cd3acf4f64d0/pexec/managed_process.go#L24

cheukt replied:
cool, that makes sense. I do think it is possible, but would be annoying to deal with the configuration.

cheukt commented on module/modmanager/manager.go:
```diff
@@ -1119,15 +1121,55 @@ func (mgr *Manager) FirstRun(ctx context.Context, conf config.Module) error {
+	go func() {
+		for scanOut.Scan() {
+			out := scanOut.Text()
+			logger.Infow("first run stdio", "output", out)
```
double check that this output gets sent to app.viam, I think we may have to use Infof so it's not lost

maximpertsov replied:
i'm going to address the app.viam bug in a separate PR - unfortunately I think it might have something to do with my `AsZap` conversation, that I needed to do in order to call the `With` method.

cheukt replied:
I think this is a blocker for merge, we should just fix it now and be done with it

maximpertsov replied:
fixed https://github.com/viamrobotics/rdk/pull/4475/commits/bfc1c466223bb6848cbee61b9122811685d21cbd

it was the `AsZap` conversion as I suspected. weirdly I got an error when I didn't do that conversion initially - maybe it's fine now because we grab a sublogger first?

maximpertsov replied:
ugh nvm, I still get a test failure when I don't convert to `AsZap`. super weird because it seems to work fine in non-test environments

sorry @dgottlieb @cheukt, I asked for a re-review prematurely. need to resolve this matter first

maximpertsov replied:
the failure is a bit hard to spot because it involves a panic, but essentially it panics because I try to call `.WithFields` which seems to hit [this logic](https://github.com/viamrobotics/rdk/blob/1930eb10ab992374f3d8206a16dc93f62a873b77/logging/logger.go#L114-L117) in tests but if fine otherwise.

maximpertsov replied:
fixed by [implemented the missing method](https://github.com/viamrobotics/rdk/pull/4475/commits/0808575a2181cfa334f3754b0a06798776a3aeb3) - confirmed that logs still go to app.viam

dgottlieb replied:
I'm a little lost, but in the end we wound up removing the `AsZap`. That seems right to me. Was the only motivation for using `AsZap` in the original because we missed the existance of [WithFields](https://github.com/viamrobotics/rdk/blob/a6e455af13cc1a85e9421e65ce64be94d7aec16b/logging/logger.go#L23)?

The [missing method](https://github.com/viamrobotics/rdk/pull/4475/commits/0808575a2181cfa334f3754b0a06798776a3aeb3) only applies to `zLogger`s and I see nothing here that is [using](https://github.com/viamrobotics/rdk/blob/a6e455af13cc1a85e9421e65ce64be94d7aec16b/logging/logger.go#L84) `FromZapCompatible`

maximpertsov replied:
So if I remove the change you reference [in this comment](https://github.com/viamrobotics/rdk/pull/4475#discussion_r1811694898), I see a panic in our tests that points the line that calls `WithFields`.

maximpertsov commented on module/modmanager/manager.go:
```diff
@@ -1082,18 +1083,19 @@ func (m *module) checkReady(ctx context.Context, parentAddr string, logger loggi
 	}
 
-	logger.Info("executing first run script")
+	logger = logger.WithFields("module", conf.Name, "path", firstRunPath)
```
apparently I need to declare the full set of fields I wish to include with this logger's logs. if I only put `path` here then `module` no longer shows up, even thought I declared in a `WithFields` call earlier.

dgottlieb replied:
Sorry for that rough edge -- I can confirm the code does do what you're describing.

maximpertsov replied:
No worries, it's a perfectly reasonable behavior - it didn't take me long to figure out what was happening.

maximpertsov commented:
one stubborn issue remains - no matter what I always get error log at the end:

```
error scanning first run stdio  {"error":"read |0: file already closed",...}
```

maximpertsov commented:
> one stubborn issue remains - no matter what I always get error log at the end:
> 
> ```
> error scanning first run stdio  {"error":"read |0: file already closed",...}
> ```

This might be unavoidable, unless my understanding is off. I decided to ignore this particular error and left a code comment as to why: https://github.com/viamrobotics/rdk/pull/4475/commits/ace32d32280c9955facb6651adb013be85d2cbd3

let me know if anyone is unhappy with this

maximpertsov commented:
@dgottlieb @cheukt I fixed the [`WithFields` issues](https://github.com/viamrobotics/rdk/pull/4475#discussion_r1811458581) and addressed a few other shortcomings - this PR is ready for review once again.

dgottlieb commented:
> This might be unavoidable, unless my understanding is off. I decided to ignore this particular error and left a code comment as to why: https://github.com/viamrobotics/rdk/commit/ace32d32280c9955facb6651adb013be85d2cbd3

> let me know if anyone is unhappy with this

That behavior is a bit surprising to me, but I'm seeing other uses of scanner doing the same, so certainly seems to be an accurate depiction of what errors get returned

dgottlieb reviewed (approved):
This looks great to me, thanks for this pass Maxim

maximpertsov commented:
> > This might be unavoidable, unless my understanding is off. I decided to ignore this particular error and left a code comment as to why: [ace32d3](https://github.com/viamrobotics/rdk/commit/ace32d32280c9955facb6651adb013be85d2cbd3)
> 
> > let me know if anyone is unhappy with this
> 
> That behavior is a bit surprising to me, but I'm seeing other uses of scanner doing the same, so certainly seems to be an accurate depiction of what errors get returned

Yeah, this what I saw other golang projects needing to do when I did a broad "scan" (sorry) of the ecosystem :disappointed:

maximpertsov commented on logging/logger.go:
```diff
@@ -111,9 +111,8 @@ func (logger *zLogger) AddAppender(appender Appender) {
-	// Not supported. Use With() instead.
-	return nil
+func (logger *zLogger) WithFields(args ...interface{}) Logger {
+	return &zLogger{logger.AsZap().With(args...)}
```
i'm going to move this discussion to [this thread](https://github.com/viamrobotics/rdk/pull/4475#discussion_r1813091766) because the context behind this change is there.

dgottlieb replied:
@maximpertsov showed me what he was observing. The `TestModuleDebugReconfigure` test is using [this helper](https://github.com/viamrobotics/rdk/blob/8e378dbf990490b584ddfab4ef3f12f67322070e/testutils/log_utils.go?plain=1#L22) which definitely uses this code.

No objection for keeping this change in.

cheukt commented on module/modmanager/manager.go:
```diff
@@ -1119,15 +1121,49 @@ func (mgr *Manager) FirstRun(ctx context.Context, conf config.Module) error {
+		logger.Errorw("failed", "error", err)
+		return err
+	}
+	logger.Infow("succeeded")
```
```suggestion
	logger.Info("first run script succeeded")
```

maximpertsov replied:
so I deliberately removed this from the log message since "first run" is indicated by the sublogger name: https://github.com/viamrobotics/rdk/pull/4475/commits/cfdd5df3afdb899384439d421f5e9acd701218a3

maximpertsov replied:
on second thought, maybe it's fine to be a teensy-bit more verbose - will update with your suggestion

maximpertsov replied:
the main log I wanted to trim "first run" from was the stdio + stderr logs, since that distracted from the subprocess output IMO

cheukt replied:
ya I think getting rid of those from the stdio + stderr logs definitely makes sense

cheukt commented on module/modmanager/manager.go:
```diff
@@ -1119,15 +1121,49 @@ func (mgr *Manager) FirstRun(ctx context.Context, conf config.Module) error {
+		return err
+	}
+	if err := cmd.Wait(); err != nil {
+		logger.Errorw("failed", "error", err)
```
```suggestion
		logger.Errorw("first run script failed", "error", err)
```

cheukt commented on module/modmanager/manager.go:
```diff
@@ -1119,15 +1121,49 @@ func (mgr *Manager) FirstRun(ctx context.Context, conf config.Module) error {
+		}
+	}()
+	if err := cmd.Start(); err != nil {
+		logger.Errorw("failed to start", "error", err)
```
```suggestion
		logger.Errorw("failed to start first run script", "error", err)
```

cheukt reviewed (approved):
LGTM, some suggestions for the logs but otherwise thanks for fixing!
